R/bundlePackage.R

```diff
@@ -0,0 +1,157 @@
+bundlePackages <- function(appDir,
+                           appMode,
+                           assetTypeName,
+                           hasParameters = FALSE,
+                           documentsHavePython = FALSE,
+                           quartoInfo = NULL,
+                           verbose = FALSE
+                           ) {
+  if (appMode %in% c("static", "tensorflow-saved-model")) {
+    return(list())
+  }
+
+  # Skip snapshotting R dependencies if an app does not use R. Some
+  # dependencies seem to be found based on the presence of Bioconductor
+  # packages in the user's environment.
+  if (!appUsesR(quartoInfo)) {
+    return(list())
+  }
+
+  # detect dependencies including inferred dependencies
+  inferredRDependencies <- inferRPackageDependencies(
+    appMode = appMode,
+    hasParameters = hasParameters,
+    documentsHavePython = documentsHavePython,
+    quartoInfo = quartoInfo
+  )
+  deps <- snapshotRDependencies(appDir, inferredRDependencies, verbose = verbose)
+
+  packageMessages <- c()
+  errorMessages <- c()
+  packages <- list()
+
+  # construct package list from dependencies
+  for (i in seq_len(nrow(deps))) {
+    name <- deps[i, "Package"]
+
+    # get package info
+    info <- as.list(deps[i, c("Source", "Repository")])
+
+    # include github package info
+    info <- c(info, as.list(deps[i, grep("Github", colnames(deps), perl = TRUE, value = TRUE)]))
+
+    # get package description; note that we need to remove the
+    # packageDescription S3 class from the object or jsonlite will refuse to
+    # serialize it when building the manifest JSON
+    # TODO: should we get description from packrat/desc folder?
+    info$description <- suppressWarnings(unclass(utils::packageDescription(name)))
+
+    # TODO(HW): should this code be removed? I don't see how it is ever reached
+    # if description is NA, application dependency may not be installed
+    if (is.na(info$description[1])) {
```

Comment:
Do you understand how we hit this code? Is it if there was a previous snapshot and those packages are no longer installed?

Reply:
Ah got it, I'll change my test.

Reply:
I've failed to create this situation in a test, so I'll just do some mocking to simulate it.